civo/kubernetes-marketplace
# use helm crd

K3s ships with a helm operator that accepts a CRD and enforces the deployment.  More information can be found [here](https://rancher.com/docs/k3s/latest/en/configuration/#auto-deploying-manifests). This has been tested in a vanilla civo cluster.


_benefits_:
- This greatly simplifies the deployment
- makes it more declarative
- avoids the case of race conditions (control loop will continue to try and reconcile if chart fails due to missing CRDs)
To test:

```bash
cat <<EOF > /tmp/app.yaml
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: prometheus-operator
  namespace: kube-system
spec:
  chart: stable/prometheus-operator
  version: 8.2.0
  targetNamespace: monitoring
EOF

kubectl apply -f /tmp/app.yaml
```

----
If your pull request concerns an existing Marketplace application, please make sure you have:

* [x] Notified the Maintainer of the application in this pull request so that they are aware of your proposal.
* [x] Outlined the changes you are proposing, and the reasons these are required (e.g. stability, compatibility with new versions of Kubernetes implementations, etc).
* [x] Tested that the application works with the proposed updates applied (including a screenshot)
